sdk/dapp-kit/src/components/connect-modal/ConnectModal.tsx

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { WalletWithRequiredFeatures } from '@mysten/wallet-standard';
+import * as Dialog from '@radix-ui/react-dialog';
+import clsx from 'clsx';
+import { useState } from 'react';
+import type { ReactNode } from 'react';
+
+import { useConnectWallet } from '../../hooks/wallet/useConnectWallet.js';
+import { BackIcon } from '../icons/BackIcon.js';
+import { CloseIcon } from '../icons/CloseIcon.js';
+import * as styles from './ConnectModal.css.js';
+import { ConnectionStatus } from './views/ConnectionStatus.js';
+import { GettingStarted } from './views/GettingStarted.js';
+import { WhatIsAWallet } from './views/WhatIsAWallet.js';
+import { WalletList } from './wallet-list/WalletList.js';
+
+type ConnectModalView = 'getting-started' | 'what-is-a-wallet' | 'connection-status';
+
+type ConnectModalProps = {
+	trigger: ReactNode;
+};
+
+export function ConnectModal({ trigger }: ConnectModalProps) {
+	const [isConnectModalOpen, setConnectModalOpen] = useState(false);
+	const [currentView, setCurrentView] = useState<ConnectModalView>();
+	const [selectedWallet, setSelectedWallet] = useState<WalletWithRequiredFeatures>();
+	const { mutate, isError } = useConnectWallet();
+
+	const connectWallet = (wallet: WalletWithRequiredFeatures) => {
+		setCurrentView('connection-status');
```

Comment:
there's the potential for a slight flash of content when connecting to a previously connected wallet which is a pretty instant operation. I'll probably set a quick timeout in a later PR